internal/bodyprocessors/raw.go

```diff
@@ -0,0 +1,40 @@
+package bodyprocessors
+
+import (
+	"io"
+	"strconv"
+	"strings"
+
+	"github.com/corazawaf/coraza/v3/experimental/plugins/plugintypes"
+	"github.com/corazawaf/coraza/v3/internal/collections"
+)
+
+type rawBodyProcessor struct {
+}
+
+func (*rawBodyProcessor) ProcessRequest(reader io.Reader, v plugintypes.TransactionVariables, options plugintypes.BodyProcessorOptions) error {
+	buf := new(strings.Builder)
+	if _, err := io.Copy(buf, reader); err != nil {
+		return err
+	}
+
+	b := buf.String()
+
+	v.RequestBody().(*collections.Single).Set(b)
+	v.RequestBodyLength().(*collections.Single).Set(strconv.Itoa(len(b)))
```

Comment:
Shall we use b.Len here? https://pkg.go.dev/strings#Builder.Len

Reply:
I have no preference, internally it's just calling `len` on the internal buffer, so no difference as far as I can tell.